Prototyping/Domains/ThemePark/Invasion/acServer.js

```diff
@@ -0,0 +1,855 @@
+// AC stands for Alien Control
+
+/* globals Entities, Agent, Script, EntityViewer, Sound, SoundCache, Avatar, Quat, Vec3, Users, Messages */
+var invasionUtils = Script.require('./invasionUtils.js');
+
+var MILLISECONDS_IN_SECOND = 1000;
+var ADDED_LIFETIME = 15;
+var LIFETIME_CHECK_INTERVAL = (ADDED_LIFETIME * MILLISECONDS_IN_SECOND) / 2;
+
+var SOUNDS_PATH = 'https://cdn.glitch.com/b1db0709-5c86-4342-8be4-19325d64d41a/sounds/';
+var INVASION_SOUND = SOUNDS_PATH + '219199__airborne80__invasion.wav';
+var ALARM_SOUND = SOUNDS_PATH + '263682__ryanconway__warning-alarm-louder.wav';
+
+var MOTHERSHIP_POSITION = {x: 15.8281, y: 84.2781, z: 10.0205};
+var BEAM_IN_OFFSET = {x: 0.0, y: -33.0, z: 0.0};
+var BEAM_FROM_POSITION = MOTHERSHIP_POSITION;
+var BEAM_TO_POSITION = Vec3.sum(MOTHERSHIP_POSITION, BEAM_IN_OFFSET);
+
+// TODO: number of UFO's depend on the amount of active-guests in the park
+var NUMBER_OF_UFOS = 20;
+
+var CLOSEST_LASER_RANGE = 10;
+var LOWEST_HOVER_HEIGHT = -3.75;
+var RANDOM_ATTACK_BOX_DIMENSIONS = {x: 12, y: 8, z: 12};
+
+var UFO_ATTACK_INTERVAL_MS = 100;
+var UFO_ATTACK_DAMAGE = 0.0015 * 3;
+
+var DEBUG_CUBES = false;
+
+var invasionSound = SoundCache.getSound(INVASION_SOUND);
+var alarmSound = SoundCache.getSound(ALARM_SOUND);
+
+var POWER_SOURCE_FULL_ANGULAR_VELOCITY = {
+    x: 0,
+    y: 0.2617993950843811,
+    z: 0
+};
+
+var POWER_SOURCE_PROPERTIES = {
+    angularDamping: 0,
+    angularVelocity: {
+        x: 0,
+        y: 0.2617993950843811,
+        z: 0
+    },
+    damping: 0,
+    dimensions: {
+        x: 4.974679946899414,
+        y: 5.103892803192139,
+        z: 8.011173248291016
+    },
+    modelURL: "atp:/0lAxOepK_-G_2fFBRNiK6y4_obj/tinker.obj",
+    name: "PowerSource",
+    rotation: {
+        w: -0.421863317489624,
+        x: 0.4218331277370453,
+        y: 0.5675212740898132,
+        z: 0.567457377910614
+    },
+    script: "https://hifi-content.s3.amazonaws.com/davidback/development/themepark/invasion/powerDevice.js",
+    scriptTimestamp: Date.now(), // invalidate cache while dev
+    type: "Model",
+    userData: JSON.stringify({
+        grabbableKey: {
+            grabbable: false
+        }
+    })
+};
+
+var ALIEN_PROPERTIES = {
+    damping: 0,
+    angularDamping: 0,
+    dimensions: {
+        x: 2.922884702682495,
+        y: 3.074074029922485,
+        z: 2.251689076423646
+    },
+    modelURL: "https://hifi-content.s3.amazonaws.com/davidback/development/themepark/invasion/alien.obj",
+    name: "UFO with Alien",
+    rotation: {
+        w: 0.8813782930374146,
+        x: 0,
+        y: -0.4724111258983612,
+        z: 0
+    },
+    script: "https://hifi-content.s3.amazonaws.com/davidback/development/themepark/invasion/alien.js",
+    scriptTimestamp: Date.now(), // invalidate cache while dev
```

Comment:
TODO: links should be relative, and dynamic time-stamp work-arounds should be removed